packages/arb-avm-cpp/data_storage/include/data_storage/readtransaction.hpp

```diff
@@ -14,29 +14,25 @@
  * limitations under the License.
  */
 
-#ifndef data_storage_readonlytransaction_hpp
-#define data_storage_readonlytransaction_hpp
+#ifndef data_storage_readtransaction_hpp
+#define data_storage_readtransaction_hpp
 
 #include <data_storage/datastorage.hpp>
 
+class ReadSnapshotTransaction;
 class ReadWriteTransaction;
 
 class ReadTransaction {
    private:
     std::unique_ptr<Transaction> transaction{};
     rocksdb::ReadOptions read_options{};
 
+    friend ReadSnapshotTransaction;
```

Comment:
Rather than making that class a friend, the variables it needs to access should just be `protected` I think

Reply:
👍